File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    answer = []
    slen = len(sequence)
    mn = 10000001
    # sequence를 순회
    tlst = []
    s,e,tmp = 0,0,sequence[0]
    while s<=e<slen:
        # cnt를 추가하면서 최소 길이 인덱스 찾기
        if tmp == k:
            if e-s+1 < mn:
                mn = e-s+1
                answer = [s,e]
            tmp -= sequence[s]
            s += 1
        elif tmp > k:
            tmp -= sequence[s]
            s += 1
        elif tmp <k:
            e += 1
            if e < slen:
                tmp += sequence[e]
    return answer

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> sequence, int k) {
    vector<int> answer(2,0);
    int s=0,e=0;
    int mn = 10000001;
    int slen = sequence.size();
    int tmp = sequence[0];
    
    while (s<=e && e<slen){
        if (tmp == k) {
            if (e-s+1 < mn) {
                mn = e-s+1;
                answer[0] = s;
                answer[1] = e;
            }
            tmp -= sequence[s];
            s++;
        }    
        else if (tmp > k) {
            tmp -= sequence[s];
            s++;
        }
        else if (tmp < k) {
            e++;
            if (e < slen) { 
                tmp += sequence[e];
            }
        }
    }
    return answer;
}